### ISSUE TITLE
Refactoring of junk code

### ISSUE THREAD

**Issue by ifedchankau** [linterhub/usage-parser#62]
Next solutions isn't good:

* function handle of handle.js file
```
const context = require('./template/context.js');
context.options = [];
```
* function templatizer of templatizer.js file
```
const argumentsTemplate = JSON.parse(fs.readFileSync('./src/template/args.json'));
argumentsTemplate.definitions.arguments.properties = {};
```

It it necessary, because javascript filling empty templates, and for each next iteration of parser, js using this filled templates. Please, find best solution, how to fix it

**Comment by ifedchankau** [comment]
Functions ```setSectionByNames```, ```setSectionWithPostfixArray```, ```setSectionWithPostfixSingle``` of handle.js have too much arguments. Please, refactor it. Maybe, it will be good to move help and user configuration to context [internal configuration].

Related #97

**Comment by ifedchankau** [comment]
In project there are too many utility functions, for example:
- `removeChar`
- `removeExtraSpaces`
- `getValueByRegexp`
- `isValueBoolean`

I propose to move all this functions to separate file.

**Comment by ifedchankau** [comment]
In our regular expressions there is no need to return all capture groups in brackets. I propose to make non-capture groups with `?:`.

For example, regexp of default value
`(\[|\(|[\s]+)default(s|)(:|[\s](to|at|is))[\s]+(.*?)(\]|\)|([\s]+|$))`
must be changed to
`(?:\[|\(|[\s]+)default(?:s|)(?::|[\s](to|at|is))[\s]+(.*?)(?:\]|\)|([\s]+|$))`

**Comment by ifedchankau** [comment]
Next solution in `section.js` in `splitOptionsSection` function for section name removing isn't good:
`string.match(/([\S]|[\S]+[\s][\S]+):$/) === null`

Please, refactor it.

**Comment by ifedchankau** [comment]
I propose to move delimiters array `[' ', '=']` and array of extra chars for deliting to context.

Alsi, i propose to remove options and delimiter from context.

**Comment by xferra** [comment]
Some of those statements are not clear without explanation why it should be refactored.

**Comment by romanrostislavovich** [comment]
Part of #170

**Comment by romanrostislavovich** [comment]
Closed in #174
